lexer.js

```diff
@@ -55,7 +55,18 @@ function parseCJS (source, name = '@') {
 function decode (name) {
   if (name[0] === '"' || name[0] === "'") {
     try {
-      return (0, eval)(name);
+      const decoded = (0, eval)(name);
+      let expectSurrogate = false;
+      for (let i = 0; i < decoded.length; i++) {
+        const surrogatePrefix = decoded.charCodeAt(i) & 0xFC00;
+        if (!expectSurrogate && surrogatePrefix !== 0xDC00 && surrogatePrefix !== 0xD800)
+          continue;
+        if (expectSurrogate && surrogatePrefix !== 0xDC00 || !expectSurrogate && surrogatePrefix !== 0xD800)
+          return;
+        expectSurrogate = !expectSurrogate;
+      }
+      if (!expectSurrogate)
+        return decoded;
```

Comment:
This can probably be simplified to check the next char code in the same iteration (I might have got `0xDC00` and `0xD800` in the wrong order):
```js
      for (let i = 0; i < decoded.length; i++) {
        const surrogatePrefix = decoded.charCodeAt(i) & 0xFC00;
        if (
          (surrogatePrefix === 0xD800 && decoded.charCodeAt(++i) & 0xFC00 !== 0xDC00) ||
          surrogatePrefix === 0xDC00
        ) return;
      }
      return decoded;
```

Reply:
Added this and also excluded invalid ranges per the spec as well. Note `(decoded.charCodeAt(++i) & 0xFC00)` requires parens there!